### PR TITLE
py-contourpy: add v1.3.0

### PR DESCRIPTION
**Pull request by adamjstewart**
Adds a newer version of py-contourpy with numpy 2 support

Depends on #46230

### CODE REVIEW

**Comment by adamjstewart** [comment]
@alalazo how did the package audit tests pass when pybind11 2.13 doesn't exist yet?

**Review by tldahlgren — APPROVED** [review]
Confirmed sha256.

File: var/spack/repos/builtin/packages/py-contourpy/package.py [hunk 1]
```diff
@@ -14,17 +14,31 @@ class PyContourpy(PythonPackage):
...
-    depends_on("python@3.7:", type=("build", "link", "run"))
-    depends_on("py-pybind11@2.6:", type=("build", "link"))
-    depends_on("py-setuptools@42:", type="build")
-    depends_on("py-numpy@1.16:", type=("build", "run"))
-    # https://github.com/numpy/numpy/issues/26191
-    depends_on("py-numpy@:1", when="@:1.2.0", type=("build", "run"))
+    with default_args(type="build"):
+        depends_on("meson@1.2:")
+        depends_on("py-meson-python@0.13.1:")
+        depends_on("py-pybind11@2.13.1:", when="@1.3:")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-contourpy/package.py:26]: We just check dependencies:

https://github.com/spack/spack/blob/636843f3308cb89458220bf10497912a9debbb16/lib/spack/spack/audit.py#L968-L975

and the condition above is satisfiable by `py-pybind11@master`

Comment (alalazo) [reply var/spack/repos/builtin/packages/py-contourpy/package.py:26]: @adamjstewart :point_up:

**Pull request merged**